### DISCUSSION TITLE
Deselect unused pins in custom shape definition?

### DISCUSSION THREAD

**Discussion by jongena** [hneemann/Digital#978] [Q&A] [answered]
First of all, congratulations with this excellent tool to analyze logic circuits !

I am building several custom circuits. One of them consists of a set of D-flip-flops where only the Q output is in use.
The Q~ outputs are not used. How can I deselect these on the custom layout?

I first tried to connect the "unconnected" icon to these pins, but this creates an error message when I try to run the simulation.  (The Unconnected symbol is not allowed here).
Next I tried to  put labelled outputs to all of these to avoid the error message (Shape xxx is missing. Found a pin without a label) when calling the shape.  But then I get a lot of not needed pins on the layout of the newly created circuit. Is there a way to avoid these and just have the minimum needed pins on the layout ? Or what is the recommended way ? I am most likely overlooking something ... ? 
Can you send me some more information ?
Thanks in advance !

**Comment by xotmatrix** [comment, marked as answer]
I put a pull-down resistor on the unused output and that seemed work. I don't know if there is a better option.

**Reply by jongena** [reply]
Many thanks for the hint !  This works !

**Reply by hneemann** [reply]
@jongena 
To be honest, I do not understand this question! Can you describe in more detail what you want to achieve?

**Comment by xotmatrix** [comment]
I confess, I'm a little confused myself. I thought I had reproduced this embedded circuit problem but it seems maybe I didn't. It's true:

* Connecting an unneeded output to the X "Not Connected" component does result in an error. 
* Connecting an unneeded output to an unlabeled output component results in an error.
* Connecting an unneeded output to a labeled output component, but removing it from a custom shape results in an error.

Of these, I don't think the first should produce an error.

However, all that said, simply leaving the unneeded output alone with nothing connected to it, prevents it from appearing in its embedded circuit shape and causes no error. I feel like I must have tried this but maybe I didn't. My pull-down resistor "solution" doesn't cause an error but I guess it isn't really needed.

**Reply by hneemann** [reply]
That's what I thought: simply not connecting anything to the output works without problems.

The "Not Connected" component is intended for connecting to unused inputs. These would otherwise generate an error, since unconnected inputs are not allowed. The "Not Connected" component is internally an output that always outputs "High Z".
So it makes little sense to connect it to an output. But you could think about whether this should generate an error message or simply be ignored.

**Reply by jongena** [reply]
Many thanks for the additional replies. The case I had involuntarily realised was no 2. I now have the circuit with unconnected outputs, that can be called without troubles, having labelled that one output (out of many) that had no name... Also thanks for the remarks on the "Not connected" component. 

In this respect, I think it remains useful to flag the user that a connection is made that is not very meaningful. Thus in case the "Not Connected" component is wrongly used, perhaps a 'warning' could be produced instead of an 'error' message ?
